### PR TITLE
Disallow setting modjoin past + in group chats.

### PR DESCRIPTION
**Pull request by HawkiePS**

### CODE REVIEW

File: commands.js [hunk 1]
```diff
@@ -526,6 +526,7 @@ exports.commands = {
 				this.addModCommand("" + user.name + " turned on modjoin.");
 			} else if (target in Config.groups) {
 				if (room.battle && !this.can('makeroom')) return;
+				if (room.isPersonal && !this.can('makeroom') && target !== '+') return this.errorReply("/" + cmd + " - Access denied from setting modjoin past + in Personal Rooms.");
 				room.modjoin = target;
 				this.addModCommand("" + user.name + " set modjoin to " + target + ".");
 			} else {
```
Comment (AustinXII) [inline commands.js]: You need to put cmd in the function

Comment (Zarel) [inline commands.js]: That's not the right line to comment on...

File: commands.js [hunk 2]
```diff
@@ -526,6 +526,7 @@ exports.commands = {
 				this.addModCommand("" + user.name + " turned on modjoin.");
 			} else if (target in Config.groups) {
 				if (room.battle && !this.can('makeroom')) return;
+				if (room.isPersonal && !this.can('makeroom') && target !== '+') return this.errorReply("/" + cmd + " - Access denied from setting modjoin past + in Personal Rooms.");
```
Comment (Zarel) [inline commands.js]: You shouldn't use `cmd` at all, just say "/modjoin -"

**Comment by Zarel** [comment]
Anyway, this is a policy decision; you need to get people to agree on it before making a pull request.

**Comment by HawkiePS** [comment]
My bad. I assumed it was never supposed to happen because it could cause problems with staff being unable to join public/hidden groupchats to solve problems reported to them. Anyhow I'll make the change and fixup then see how people feel about it. Noted for the future.

**Comment by kotarou3** [comment]
Staff with the `makeroom` permission can override modjoin

**Comment by HawkiePS** [comment]
The justification is the same for battle rooms with modjoin restricted to +, so any staff member can deal with issues that arise

**Comment by Zarel** [comment]
Oh, that makes sense; I understand your reasoning now.

File: commands.js [hunk 3]
```diff
@@ -526,6 +526,7 @@ exports.commands = {
 				this.addModCommand("" + user.name + " turned on modjoin.");
 			} else if (target in Config.groups) {
 				if (room.battle && !this.can('makeroom')) return;
+				if (room.isPersonal && !this.can('makeroom') && target !== '+') return this.errorReply("/modjoin - Access denied from setting modjoin past + Personal Rooms.");
```
Comment (Zarel) [inline commands.js]: `this.can` gives an error message. If you want to replace that error message, you need to use `user.can`

**Comment by HawkiePS** [comment]
Updated.

File: commands.js (hunk 1 above)
Comment (AustinXII) [inline commands.js]: I did it on my phone so it was kinda laggy :/

File: commands.js [hunk 4]
```diff
@@ -526,6 +526,7 @@ exports.commands = {
 				this.addModCommand("" + user.name + " turned on modjoin.");
 			} else if (target in Config.groups) {
 				if (room.battle && !this.can('makeroom')) return;
+				if (room.isPersonal && !user.can('makeroom') && target !== '+') return this.errorReply("/modjoin - Access denied from setting modjoin past + Personal Rooms.");
```
Comment (Zarel) [inline commands.js]: There's a difference between names used in the code and names used to talk to users.

`isPersonal` is only what they're called in the code. In reality, we call them "group chats", not "Personal Rooms".

Also, "past + group chats" is grammatically weird. Try "past + in group chats".

**Pull request merged**